### ISSUE TITLE
[BUG] `rancher2_cluster_v2` field `cluster_id` does not exists

### ISSUE THREAD

**Issue by Wykiki** [rancher/terraform-provider-rancher2#1104]
### Rancher Server Setup
- Rancher version:
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

### Information about the Cluster
- Kubernetes version:
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

### User Information
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

### Provider Information
- What is the version of the Rancher v2 Terraform Provider in use? 1.25.0
- What is the version of Terraform in use? 1.4.4

### Describe the bug

The attribute `rancher2_cluster_v2.cluster_id` is not available, while it's mentionned in the documentation [here](https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/cluster_v2#cluster_id).

The following attribute works, but is not documented : `rancher2_cluster_v2.id`

### To Reproduce

```tf
terraform {
  required_providers {
    rancher2 = {
      source  = "rancher/rancher2"
      version = ">= 1.25"
    }
  }
}

resource "rancher2_cluster_v2" "cluster" {
  name               = "test"
  kubernetes_version = "1.2.3"
}

resource "rancher2_app_v2" "app" {
  cluster_id    = rancher2_cluster_v2.cluster.cluster_id
  name          = "test"
  namespace     = "test"
  repo_name     = "test"
  chart_name    = "test"
}
```

### Actual Result

```
❯ terraform validate
╷
│ Error: Unsupported attribute
│ 
│   on rancher.tf line 16, in resource "rancher2_app_v2" "app":
│   16:   cluster_id    = rancher2_cluster_v2.cluster.cluster_id
│ 
│ This object has no argument, nested block, or exported attribute named "cluster_id".
```

### Expected Result

```
❯ terraform validate
Success! The configuration is valid.
```

### Additional context

Following configuration works :

```tf
terraform {
  required_providers {
    rancher2 = {
      source  = "rancher/rancher2"
      version = ">= 1.25"
    }
  }
}

resource "rancher2_cluster_v2" "cluster" {
  name               = "test"
  kubernetes_version = "1.2.3"
}

resource "rancher2_app_v2" "app" {
  cluster_id    = rancher2_cluster_v2.cluster.id
  name          = "test"
  namespace     = "test"
  repo_name     = "test"
  chart_name    = "test"
}
```

Output :

```
❯ terraform validate
Success! The configuration is valid.
```

Related to rancher/terraform-provider-rancher2#1092

@a-blender @HarrisonWAffel

**Comment by a-blender** [comment]
@Wykiki Thank you for this, but I just see you performing a local validate. I believe we were asking what your "use case" is for this, so what infra are you using if any? What type of cluster? And do both the `id` and `cluster_id` fields behave the same way for that validation case.

There doesn't need to be both `id` and `cluster_id` fields in the provider if they are both being used to reference the id of the downstream cluster. In that case, I'm thinking `cluster_id` should be removed from the schema but this needs to be tested.

**Comment by Wykiki** [comment]
TLDR; I misread the documentation, I guess we can close the issue / PR

@a-blender I tried to use it to create `rancher2_app_v2` resource, in a RKE2 cluster.  
Digging up a bit, I ended up using the field `cluster_v1_id`, because the field `cluster_id` doesn't exist in a RKE2 cluster resource. It is in fact an attribute of the `cluster_registration_token` if I understand correctly.  
Also, the `id` field didn't contain appropriate ID that I could use in `rancher2_catalog_v2` or `rancher2_app_v2` resources.

Example of working configuration for my usecase :

```tf
terraform {
  required_providers {
    rancher2 = {
      source  = "rancher/rancher2"
      version = ">= 1.25"
    }
  }
}

resource "rancher2_cluster_v2" "cluster" {
  name               = "test"
  kubernetes_version = "1.2.3"
}

resource "rancher2_catalog_v2" "spark_operator" {
  cluster_id = rancher2_cluster_v2.cluster.cluster_v1_id
  name       = "spark-operator"
  url        = "https://googlecloudplatform.github.io/spark-on-k8s-operator"
}

resource "rancher2_app_v2" "spark_operator" {
  cluster_id    = rancher2_cluster_v2.cluster.cluster_v1_id
  name          = "spark-operator"
  namespace     = "spark-operator"
  repo_name     = rancher2_catalog_v2.spark_operator.name
  chart_name    = "spark-operator"
  chart_version = "1.1.27"
  values = yamlencode({
  })
}
```

**Comment by HarrisonWAffel** [comment]
@Wykiki Thanks for updating this issue and providing a working example, @a-blender I'm fine with closing this out if you are too.

**Comment by a-blender** [comment]
@HarrisonWAffel Should their use case be verified by QA?

**Comment by HarrisonWAffel** [comment]
@a-blender You may want to check with QA directly about that. If Wykiki was able to rectify the issue and found that it was just a misunderstanding with the documentation I'm not sure there is anything to verify. I would be okay with a verification pass, but we should check if QA thinks its worth the time.

**Comment by a-blender** [comment]
@sowmyav27 The customer here had a docs misunderstanding and was able to fix the issue on their end without merging any PRs. I think it's good to close, but up to you. I moved To Test

**Comment by slickwarren** [comment]
closing this issue based on the above comments.
